src/Generator/Resolver/Value/Chainable/FixtureMethodCallReferenceResolver.php

```diff
@@ -0,0 +1,115 @@
+use Nelmio\Alice\Throwable\Exception\Generator\Resolver\ResolverNotFoundExceptionFactory;
+use Nelmio\Alice\Throwable\Exception\Generator\Resolver\UnresolvableValueException;
+use Nelmio\Alice\FixtureInterface;
+use Nelmio\Alice\Generator\GenerationContext;
+use Nelmio\Alice\Generator\ResolvedFixtureSet;
+use Nelmio\Alice\Generator\ResolvedValueWithFixtureSet;
+use Nelmio\Alice\Generator\Resolver\Value\ChainableValueResolverInterface;
+use Nelmio\Alice\Generator\ValueResolverAwareInterface;
+use Nelmio\Alice\Generator\ValueResolverInterface;
+use Nelmio\Alice\IsAServiceTrait;
+use Nelmio\Alice\Throwable\Exception\Generator\Resolver\UnresolvableValueExceptionFactory;
+
+final class FixtureMethodCallReferenceResolver implements ChainableValueResolverInterface, ValueResolverAwareInterface
+{
+    use IsAServiceTrait;
+
+    /**
+     * @var ValueResolverInterface
+     */
+    private $argumentResolver;
+
+    public function __construct(ValueResolverInterface $resolver = null)
+    {
+        $this->argumentResolver = $resolver;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function withValueResolver(ValueResolverInterface $resolver): self
+    {
+        return new self($resolver);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function canResolve(ValueInterface $value): bool
+    {
+        return $value instanceof FixtureMethodCallValue;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param FixtureMethodCallValue $value
+     *
+     * @throws NoSuchPropertyException
+     * @throws UnresolvableValueException
+     */
+    public function resolve(
+        ValueInterface $value,
+        FixtureInterface $fixture,
+        ResolvedFixtureSet $fixtureSet,
+        array $scope,
+        GenerationContext $context
+    ): ResolvedValueWithFixtureSet
+    {
+        if (null === $this->argumentResolver) {
+            throw ResolverNotFoundExceptionFactory::createUnexpectedCall(__METHOD__);
+        }
+
+        $functionCall = $value->getFunctionCall();
+        $arguments = $functionCall->getArguments();
+        foreach ($arguments as $index => $argument) {
+            if ($argument instanceof ValueInterface) {
+                $resolvedSet = $this->argumentResolver->resolve($argument, $fixture, $fixtureSet, $scope, $context);
+
+                $arguments[$index] = $resolvedSet->getValue();
+                $fixtureSet = $resolvedSet->getSet();
+            }
+        }
+
+        $context->markAsNeedsCompleteGeneration();
+        $fixtureReferenceResult = $this->argumentResolver->resolve($value->getReference(), $fixture, $fixtureSet, $scope, $context);
+        $context->unmarkAsNeedsCompleteGeneration();
+
+        /** @var ResolvedFixtureSet $fixtureSet */
+        list($instance, $fixtureSet) = [$fixtureReferenceResult->getValue(), $fixtureReferenceResult->getSet()];
+
+        if (! method_exists($instance, $functionCall->getName())) {
```

Comment:
I think this check could be avoided or rather done in the catch block below. That way you only check when there's an error hence there is not performance impact for the regular user, and the for the one who did a mistake it stays as user friendly